### PR TITLE
configurable home page

### PR DESCRIPTION
**Pull request by brindy**
Task/Issue URL: 
* Configurable home page - https://app.asana.com/0/392891325557410/1154997199836817
* Fix misaligned icons - https://app.asana.com/0/414709148257752/1144393195163118

Tech Design URL:
CC:

**Description**:

Split out the config for the home page in to two settings:

* New Tab -> Default or Centered Search and Favorites toggle
* Keyboard -> App Launch and New Tab

New installs now get Default layout and Favorites enabled by default.

When there's no favorites the centered search is static/fixed.

Bookmarks always shows favorites.

Also fixes misaligned icons when starting the app in landscape mode.

**Steps to test this PR**:
1. Test migration.  Existing settings should map to the same experience when updating.
1. Fresh install check default settings
1. Add favorites check UI behaves as expected
1. Check keyboard options behave as expected

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Core/UserDefaultsPropertyWrapper.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+//
+
+import Foundation
+
+// Inspired by https://swiftsenpai.com/swift/create-the-perfect-userdefaults-wrapper-using-property-wrapper/
+
+// enums are not supported on iOS 12 due to a bug in JSONEncoder, so just use primitive types or NSCodables
+
+@propertyWrapper
+public struct UserDefaultsWrapper<T> {
```
Comment (tomasstrba) [inline Core/UserDefaultsPropertyWrapper.swift:27]: 👏👏👏 👌

**Review by tomasstrba — COMMENTED** [review]
@brindy code looks fine to me. 👍  Are you planning to add any commit or can I test it?

**Comment by brindy** [comment]
@tomasstrba / @bwaresiak  - assuming the tests pass, this should be good for a final review.  Thanks in advance!

File: DuckDuckGo/HomePageSettings.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+}
+
+protocol HomePageSettings {
+    
+    var layout: HomePageLayout { get set }
+ 
+    var favorites: Bool { get set }
+
+    func migrate(from appSettigs: inout AppSettings)
```
Comment (bwaresiak) [inline DuckDuckGo/HomePageSettings.swift]: I'd consider `protocol AppSettings: AnyObject` over `inout` parameter, but no strong preference here. :)

**Review by bwaresiak — APPROVED** [review]
Good job! One optional comment inline :)

**Comment by brindy** [comment]
Thanks @bwaresiak !

**Comment by tomasstrba** [comment]
Just finished testing, good job Brindy! 👏 Appreciate you took a look on UserDefaultsWrapper

**Comment by brindy** [comment]
Thanks @tomasstrba  :)

**Pull request merged**